das/daser_test.go

```diff
@@ -251,6 +251,58 @@ func TestDASer_catchUp_fails(t *testing.T) {
 	require.ErrorIs(t, result.err, share.ErrNotAvailable)
 }
 
+func TestDASerState(t *testing.T) {
+	ds := ds_sync.MutexWrap(datastore.NewMapDatastore())
+	bServ := mdutils.Bserv()
+
+	// 30 headers in the past and 30 headers in the future (so final sample height should be 60)
+	mockGet, shareServ, sub := createDASerSubcomponents(t, bServ, 30, 30)
+	expectedFinalSampleHeight := uint64(60)
+	expectedFinalSampleWidth := uint64(len(sub.Headers[29].DAH.RowsRoots))
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	t.Cleanup(cancel)
+
+	daser := NewDASer(shareServ, sub, mockGet, ds)
+	err := daser.Start(ctx)
+	require.NoError(t, err)
+	defer func() {
+		// wait for all "future" headers to be sampled
+		for {
+			select {
+			case <-ctx.Done():
+				t.Fatal(ctx.Err())
+			default:
+				if daser.LatestSampledHeight() == expectedFinalSampleHeight {
```

Comment:
While it is still more or less fine for a test to do a busy loop(wastes CPU constantly), the real implementation should not be doing this and there should be a way to _wait_ for the desired height to be dased.

Reply:
This would require a das sub similar to how there is a heightsub.

The problem is, from a user standpoint, we already have the ability to check if a height has been dased via `CacheAvailability` and the endpoint that I've been working on here #802 

If you think this is really necessary beyond testing, I'm happy to make an issue, but for now, I wouldn't see why we need to wait for DASer to get to a certain height. For example, if a user/rollup wants to see if block at height 5000 is available and daser is still catching up at height 3400 or something, they wouldn't want to sit and wait for daser to catch up to 5000, they would just call `SharesAvailable` under the hood of the endpoint provided in #802, which will check if shares available locally, and if not, will trigger a share sampling process on the header 5000.

Reply:
I'll make an issue for this as a potential enhancement but it's not high priority bc it's only testing utility for now.